### DISCUSSION TITLE
How to use map_values

### DISCUSSION THREAD

**Discussion by MikaelElkiaer** [mikefarah/yq#1863] [Q&A] [answered]
I am having some trouble with the use of `map_values`.
One thing I don't understand, and have come to suspect is a bug, is an apparent loss of keys.

I expected this query to be equivalent to an identity function, but the result surprises me.

```bash
$ yq 'map_values(.)' <<< "{a: {b: c}}"
{a: {}}
```

However, by adding a seemingly superfluous `| to_entries | from_entries` it behaves as I would expect.

```bash
$ yq 'map_values(. | to_entries | from_entries)' <<< "{a: {b: c}}"
{a: {b: c}}
```

**Comment by MikaelElkiaer** [comment]
I have a simple task I want to do with this - create a docker-compose.override.yml with just the image keys.

I can do this by

```bash
yq '. | with_entries(select(.key=="services")) | .services as $s | .services = $s | map_values(. | to_entries | from_entries | map_values(pick(["image"])))' docker-compose.yml > docker-compose.override.yml
```

but both the `|to_entries|from_entries` as well as the `as $s` parts seem like they should not be necessary here.

**Reply by mikefarah** [reply, marked as answer]
Yep that is a bug in an older version of `yq` - please grab the latest and try again 👍🏼

**Reply by MikaelElkiaer** [reply]
Awesome, thanks!
